Fruitties/shared/src/commonMain/kotlin/com/example/fruitties/di/AppContainer.kt

```diff
@@ -31,4 +38,25 @@ class AppContainer(
             scope = CoroutineScope(Dispatchers.Default + SupervisorJob()),
         )
     }
-}
+
+    val mainViewModelFactory = viewModelFactory {
+        initializer {
+            MainViewModel(repository = dataRepository)
+        }
+    }
+
+    val cartViewModelFactory = viewModelFactory {
+        initializer {
+            CartViewModel(repository = dataRepository)
+        }
+    }
+
+    val fruittieViewModelFactory = viewModelFactory {
+        initializer {
```

Comment:
Add a comment to help show that get() is operating on CreationExtras.

        initializer { // this: CreationExtras

Reply:
Used `this[key]` instead as well